src/ctapipe/image/muon/fitting.py

```diff
@@ -50,22 +66,76 @@ def kundu_chaudhuri_circle_fit(x, y, weights):
         np.sum(weights * ((center_x - x) ** 2 + (center_y - y) ** 2)) / weights_sum
     )
 
-    return radius, center_x, center_y
+    if nan_errors_flag:
+        radius_err = np.nan
+        center_x_err = np.nan
+        center_y_err = np.nan
+    else:
+        radius_err, center_x_err, center_y_err = naive_circle_fit_error_calculator(
+            x, y, weights, radius, center_x, center_y
+        )
+
+    return radius, center_x, center_y, radius_err, center_x_err, center_y_err
 
 
-def taubin_circle_fit(x, y, mask):
+def taubin_circle_fit(
+    x, y, mask, weights=None, r_initial=None, xc_initial=None, yc_initial=None
+):
     """
-    reference : Barcelona_Muons_TPA_final.pdf (slide 6)
+    Perform a Taubin circle fit with weights (optional).
+
+    The minimized loss function in this method tends to
+    maximize the radius of the ring, whereas using a simple
+    ring equation systematically results in a smaller radius.
+    Adding weights mitigates both effects and yields a more accurate fit.
 
     Parameters
     ----------
-    x: array-like or astropy quantity
-        x coordinates of the points
-    y: array-like or astropy quantity
-        y coordinates of the points
-    mask: array-like boolean
-        true for pixels surviving the cleaning
+    x : array-like or astropy.units.Quantity
+        x-coordinates of the points.
+    y : array-like or astropy.units.Quantity
+        y-coordinates of the points.
+    mask : array-like of bool
+        Boolean mask indicating which pixels survive the cleaning process.
+    weights : array-like of float, optional
+        Weights for the points. If not provided, all points are assigned equal weights (1).
+    r_initial : astropy.units.Quantity, optional
+        Initial guess for the radius of the circle. If not provided, it defaults to 1.1 deg.
+        1.1 deg. is the approximate Cherenkov photon angle produced by muons in the atmosphere
+        at La Palma altitude (2426 m a.s.l.), with momentum greater than 15 GeV.
+    xc_initial : astropy.units.Quantity, optional
+        Initial guess for the x-coordinate of the circle center. Defaults to 0.
+    yc_initial : astropy.units.Quantity, optional
+        Initial guess for the y-coordinate of the circle center. Defaults to 0.
+    Returns
+    -------
+    radius : astropy.units.Quantity
+        Fitted radius of the circle.
+    center_x : astropy.units.Quantity
+        Fitted x-coordinate of the circle center.
+    center_y : astropy.units.Quantity
+        Fitted y-coordinate of the circle center.
+    radius_err : astropy.units.Quantity
+        Fitted radius of the circle error.
+    center_x_err : astropy.units.Quantity
+        Fitted x-coordinate of the circle center error.
+    center_y_err : astropy.units.Quantity
+        Fitted y-coordinate of the circle center error.
+
+    Raises
+    ------
+    OptionalDependencyMissing
+        If the `iminuit` package is not installed.
```

Comment:
Just use ``iminuit`` instead of nitpick ignoring it. If the link doesn't work, there is no point in trying to include it.